Source/AliveLib/Scrab.cpp

```diff
@@ -627,8 +628,483 @@ __int16 Scrab::AI_Patrol_0_4AA630()
 
 __int16 Scrab::AI_ChasingEnemy_1_4A6470()
 {
-    NOT_IMPLEMENTED();
-    return 0;
+    __int16 result;
+
+    Scrab *pScrabToFight = Scrab::FindScrabToFight_4A4E20();
+    if (pScrabToFight)
+    {
+        SetBrain(&Scrab::AI_Fighting_2_4A5840);
+        field_124_fight_target_obj_id = pScrabToFight->field_8_object_id;
+        field_108_next_motion = eScrabMotions::M_Stand_0_4A8220;
+        result = 0;
+    }
+    else
+    {
+        BaseGameObject *pObj = sObjectIds_5C1B70.Find_449CF0(field_120_obj_id);
+        BaseAliveGameObject *v6 = (BaseAliveGameObject *)pObj;
+        if (!pObj || field_6_flags.Get(BaseGameObject::eDead) || (signed int)sGnFrame_5C1B84 > field_14C && !Scrab::CanSeeAbe_4A51A0((BaseAliveGameObject *)pObj))
+        {
+            field_120_obj_id = -1;
+            field_108_next_motion = eScrabMotions::M_Stand_0_4A8220;
+            Scrab::ToPatrol_4AA600();
+            result = 0;
+        }
+        else if (v6->field_114_flags.Get(Flags_114::e114_Bit8_bInvisible))
+        {
+            field_120_obj_id = -1;
+            field_108_next_motion = eScrabMotions::M_HowlBegin_26_4A9DA0;
+            Scrab::ToPatrol_4AA600();
+            result = 9;
+        }
+        else
+        {
+            if (Scrab::CanSeeAbe_4A51A0(v6))
+            {
+                field_14C = sGnFrame_5C1B84 + field_148_attack_duration;
+            }
+            if (Event_Is_Event_In_Range_422C30(kEventAbeOhm, field_B8_xpos, field_BC_ypos, -1) && field_11C_sub_state != 26)
+            {
+                field_108_next_motion = eScrabMotions::M_HowlBegin_26_4A9DA0;
+                result = 4;
+            }
+            else
+            {
+                pObj = sObjectIds_5C1B70.Find_449CF0(field_110_id);

```

Comment:
rather than assigning to pObj just static cast directly to the assignment of pLift